specification/compiler/model/metamodel.ts

```diff
@@ -179,6 +196,8 @@ export class Interface extends BaseType {
    */
   attachedBehaviors?: string[]
   properties: Property[]
+  /** Identify containers */
+  variants?: ExternalTag
```

Comment:
Variants on interfaces can now only be used for containers. Even if from the json perspective the tag is _external_ to the data structure, it's a bit weird to have `@variants external` on a type that lists all variants (properties are _internal_ to the type).

So what about a more explicit syntax in the spec:
```ts
/** @variants container */
class QueryContainer {
  bool?: BoolQuery
  boosting?: BoostingQuery
  common?: SingleKeyDictionary<Union<CommonTermsQuery, string>>
  constant_score?: ConstantScoreQuery
  ...
```

And then in the metamodel:

```ts
type Variants = ExternalTag | InternalTag | Container

class Container {
  kind: 'container'
}

export class Interface extends BaseType {
  /** Identify containers */
  variants?: Container 
  ...

export class Union extends BaseType {
  /** Identify typed_key unions (external) and variant inventories (internal) */
  variants?: ExternalTag | InternalTag
  ...
```

This could also be a simple boolean flag, but making it part of `Variant` ensures this type lists all possible variations of the `@variants` annotation.

And ultimately that will allow us to drop the technical `Container` suffix that is now redundant and have a nice `Query` type 😉

Reply:
> And ultimately that will allow us to drop the technical Container suffix that is now redundant and have a nice Query type 😉

All for removing [smurfisms](https://devcards.io/smurf-naming-convention) but the container is not the query. It's the thing carrying the query. 

```
query
-- terms: Terms query
```

Is also confusing I think. 

Gets even more ambiguous with aggregations and nesting.

Reply:
+1 on explicitly tagging it as variant though

Reply:
Agreed!